drivers/storage/portworx/portworx.go

```diff
@@ -663,14 +661,39 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster) {
 		}
 	}
 
+	// Check if feature gate is set. If it is, honor the flag here and remove it.
+	csiFeatureFlag, featureGateSet := toUpdate.Spec.FeatureGates[string(pxutil.FeatureCSI)]
+	if featureGateSet {
+		csiEnabled, err := strconv.ParseBool(csiFeatureFlag)
+		if err != nil {
+			csiEnabled = true
+		}
+
+		// Feature flag set, but CSI spec is not defined. In this case,
+		// we want to use the feature flag value in the CSI spec.
+		if toUpdate.Spec.CSI == nil {
+			toUpdate.Spec.CSI = &corev1.CSISpec{
+				Enabled: csiEnabled,
+			}
+		} else {
+			// Set CSI spec as enabled if it's already enabled OR the flag is true.
+			toUpdate.Spec.CSI.Enabled = toUpdate.Spec.CSI.Enabled || csiEnabled
+		}
+
+		// Always remove the feature flag. Clear the feature gates map if none are set.
+		delete(toUpdate.Spec.FeatureGates, string(pxutil.FeatureCSI))
+		if len(toUpdate.Spec.FeatureGates) == 0 {
+			toUpdate.Spec.FeatureGates = nil
+		}
+	}
+
+	// Enable CSI if running in k3s environment or
+	// if this is a new Portworx installation
 	if t.isK3s || toUpdate.Spec.Version == "" {
-		// Enable CSI if running in k3s environment or
-		// if this is a new Portworx installation
-		if _, ok := toUpdate.Spec.FeatureGates[string(pxutil.FeatureCSI)]; !ok {
-			if toUpdate.Spec.FeatureGates == nil {
-				toUpdate.Spec.FeatureGates = make(map[string]string)
+		if toUpdate.Spec.CSI == nil {
```

Comment:
Just to be cautious, let's set the flag outside. In case someone has created the csi spec but not enabled the flag.
```
if toUpdate.Spec.CSI == nil {
  toUpdate.Spec.CSI = &corev1.CSISpec{}
}
toUpdate.Spec.CSI.Enabled = true
```